### PR TITLE
[KIECLOUD-137] - Add the RHPAM/DM JMS support to APB

### PR DESCRIPTION
**Pull request by spolti**
Plus, update the images version to 7.4:1.0

Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

File: modules/rhpam-apb/roles/deploy-kieserver/README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Deploys Red Hat Process Automation Manager - KIE Process Server
 * Route
 * Service
 * Ping Service
-* Secret containing keystore (if it didn't exist)
+* Secret containing keystore (if it don't exist)
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-kieserver/README.md]: doesn't

File: modules/rhpam-apb/roles/deploy-kieserver/defaults/main.yml [hunk 2]
```diff
@@ -77,3 +77,20 @@ kieserver_github_webhook_secret: "{{ apb_kieserver_github_webhook_secret | defau
...
+kieserver_jms_queue_response: "{{ apb_kieserver_jms_queue_response | default('queue/KIE.SERVER.RESPONSE', true) | trim }}"
+kieserver_jms_enable_signal: "{{ apb_kieserver_jms_enable_signal }}"
+kieserver_jms_queue_signal: "{{ apb_kieserver_jms_queue_signal | default('queue/KIE.SERVER.SIGNAL', true) | trim }}"
+kieserver_jms_enable_audit: "{{ apb_kieserver_jms_enable_audit }}"
+kieserver_jms_queue_audit: "{{ apb_kieserver_jms_queue_audit | default('queue/KIE.SERVER.AUDIT', true) | trim }}"
+kieserver_jms_audit_transacted: "{{ apb_kieserver_jms_audit_transacted }}"
+kieserver_jms_username: "{{ apb_kieserver_jms_username | default('amqUser') | trim }}"
+kieserver_jms_password: "{{ apb_kieserver_jms_password | default('amqPassword') | trim }}"
+# if empty AMQ Queues will be set on the deploy-amq/tasks/main.yml
+kieserver_jms_amq_queues: "{{ apb_kieserver_jms_amq_queues | default('', true) | trim }}"
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-kieserver/defaults/main.yml]: new line

File: modules/rhpam-apb/roles/deploy-kieserver/defaults/main.yml [hunk 3]
```diff
@@ -77,3 +77,20 @@ kieserver_github_webhook_secret: "{{ apb_kieserver_github_webhook_secret | defau
...
+kieserver_jms_executor_transacted: "{{ apb_kieserver_jms_executor_transacted  }}"
+kieserver_jms_queue_request: "{{ apb_kieserver_jms_queue_request | default('queue/KIE.SERVER.REQUEST', true) | trim }}"
+kieserver_jms_queue_response: "{{ apb_kieserver_jms_queue_response | default('queue/KIE.SERVER.RESPONSE', true) | trim }}"
+kieserver_jms_enable_signal: "{{ apb_kieserver_jms_enable_signal }}"
+kieserver_jms_queue_signal: "{{ apb_kieserver_jms_queue_signal | default('queue/KIE.SERVER.SIGNAL', true) | trim }}"
+kieserver_jms_enable_audit: "{{ apb_kieserver_jms_enable_audit }}"
+kieserver_jms_queue_audit: "{{ apb_kieserver_jms_queue_audit | default('queue/KIE.SERVER.AUDIT', true) | trim }}"
+kieserver_jms_audit_transacted: "{{ apb_kieserver_jms_audit_transacted }}"
+kieserver_jms_username: "{{ apb_kieserver_jms_username | default('amqUser') | trim }}"
+kieserver_jms_password: "{{ apb_kieserver_jms_password | default('amqPassword') | trim }}"
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-kieserver/defaults/main.yml]: shouldn't this be generated password?

File: modules/rhpam-apb/roles/deploy-amq/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    image_namespace: "{{ amq_image_namespace }}"
+    image_name: "{{ amq_image_name }}"
+    image_tag: "{{ amq_image_tag }}"
+  when: state == "present"
+
+- name: "[AMQ] Configure AMQ queues"
+  set_fact:
+    kieserver_jms_amq_queues: "{{ kieserver_jms_queue_executor }},{{ kieserver_jms_queue_request }},{{ kieserver_jms_queue_response}},{{ kieserver_jms_queue_signal }},{{ kieserver_jms_queue_audit }}"
+  when:
+    apb_kieserver_jms_amq_queues|trim == "" or
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-amq/tasks/main.yml]: `kieserver_jms_amq_queues`  should have the value passed in when calling the role from kieserver role. Try not to use apb_* vars at this point. Maybe you can set this default value using the `default` filter in kieserver/defaults/main.yml file

File: modules/rhpam-apb/roles/deploy-amq/tasks/route.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        port:
+          targetPort: amq-jolokia
+        to:
+          kind: Service
+          name: '{{ amq_service_name }}'
+  register: route_raw_amq
+
+- name: "[AMQ-JOLOKIA-CONSOLE] set route host"
+  set_fact:
+    sr_host: "{{ route_raw_amq.result.spec.host }}"
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-amq/tasks/route.yml]: Where is this used?

File: modules/rhpam-apb/roles/deploy-amq/defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+amq_image_name: amq-broker-72-openshift
+amq_image_tag: "1.2"
+amq_image_namespace: "{{ image_stream_namespace }}"
```
[the remark this answers is not in the document]
Comment (spolti) [reply modules/rhpam-apb/roles/deploy-amq/defaults/main.yml]: Just a curiosity, why is this new line needed at the end of the file?

File: modules/rhpam-apb/roles/deploy-amq/tasks/route.yml (hunk 5 above)
Comment (spolti) [reply modules/rhpam-apb/roles/deploy-amq/tasks/route.yml]: This route will expose the AMQ console.

File: modules/rhpam-apb/roles/deploy-kieserver/defaults/main.yml (hunk 3 above)
Comment (spolti) [reply modules/rhpam-apb/roles/deploy-kieserver/defaults/main.yml]: +1

File: modules/rhpam-apb/roles/deploy-amq/tasks/route.yml (hunk 5 above)
Comment (ruromero) [reply modules/rhpam-apb/roles/deploy-amq/tasks/route.yml]: I see you are creating the route but why you need to save create the fact `sr_host` with the host?

Comment (spolti) [reply modules/rhpam-apb/roles/deploy-amq/tasks/route.yml]: Ah I see your point, yes, not needed.

File: modules/rhpam-apb/roles/deploy-amq/defaults/main.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ruromero) [reply modules/rhpam-apb/roles/deploy-amq/defaults/main.yml]: Text files in unix should end with a new line. It's an old discussion that ended up being a good habit

[the remark this answers is not in the document]
Comment (spolti) [reply modules/rhpam-apb/roles/deploy-amq/defaults/main.yml]: Good to know, thanks.

File: modules/rhpam-apb/roles/deploy-amq/tasks/main.yml (hunk 4 above)
Comment (spolti) [reply modules/rhpam-apb/roles/deploy-amq/tasks/main.yml]: I tried to add the default queues names on kieserver/defaults/main.yml but didn't work very well thus  I created a task for setting it.

**Comment by spolti** [comment]
@ruromero can you please review again?

File: modules/rhpam-apb/roles/deploy-kieserver/tasks/main.yml [hunk 7]
```diff
@@ -90,5 +103,6 @@
       k8s:
         state: '{{ state }}'
         definition: "{{ lookup('template', 'kieserver-dc.yml.j2') | from_yaml }}"
+      register: k8s_result
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-kieserver/tasks/main.yml]: is this registered fact used anywhere?

Comment (spolti) [reply modules/rhpam-apb/roles/deploy-kieserver/tasks/main.yml]: It is probably some garbage left from my debugging.

File: modules/rhpam-apb/roles/deploy-amq/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    image_namespace: "{{ amq_image_namespace }}"
+    image_name: "{{ amq_image_name }}"
+    image_tag: "{{ amq_image_tag }}"
+  when: state == "present"
+
+- name: "[AMQ] Configure AMQ queues"
+  set_fact:
+    kieserver_jms_amq_queues: "{{ kieserver_jms_queue_executor }},{{ kieserver_jms_queue_request }},{{ kieserver_jms_queue_response}},{{ kieserver_jms_queue_signal }},{{ kieserver_jms_queue_audit }}"
+  when:
+    kieserver_jms_amq_queues is undefined
```
Comment (ruromero) [inline modules/rhpam-apb/roles/deploy-amq/tasks/main.yml]: As you are assigning a default value, this should be
```
kieserver_jms_amq_queues == ""
```

**Pull request merged**